providers/provider_data.go

```diff
@@ -197,10 +197,6 @@ func (p *ProviderData) setProviderDefaults(defaults providerDefaults) {
 	if p.Scope == "" {
 		p.Scope = defaults.scope
 	}
-
```

Comment:
Why not make sure every provider is using `setProviderDefaults`? Does every provider definitely use `buildSessionFromClaims`?

Moving it to `buildSessionFromClaims` means the check is done on every authentication request rather than just in set up, so there's an efficiency hit too